### PR TITLE
fix d-vector

### PR DESCRIPTION
**Pull request by WeberJulian**

### CODE REVIEW

File: TTS/tts/datasets/dataset.py [hunk 1]
```diff
@@ -415,7 +415,7 @@ def collate_fn(self, batch):
 
             # get pre-computed d-vectors
             if self.d_vector_mapping is not None:
-                wav_files_names = [batch["wav_file_name"][idx] for idx in ids_sorted_decreasing]
+                wav_files_names = list(batch["wav_file_name"])
```
Comment (WeberJulian) [inline TTS/tts/datasets/dataset.py:418]: batch is already sorted

File: TTS/tts/utils/synthesis.py [hunk 2]
```diff
@@ -173,6 +173,7 @@ def embedding_to_torch(d_vector, cuda=False):
     if d_vector is not None:
         d_vector = np.asarray(d_vector)
         d_vector = torch.from_numpy(d_vector).type(torch.FloatTensor)
+        d_vector = d_vector.squeeze().unsqueeze(0)
```
Comment (WeberJulian) [inline TTS/tts/utils/synthesis.py:176]: enforce [1,D] shape, caused issue with test sentences in glowTTS

**Comment by WeberJulian** [comment]
Should I add a `train_glow_tts_d-vector.py` recipe? If yes, does it means I need to resample and compute embedding in a subprocess in the recipe? Or have a main bash script that successively call resample -> compute emb -> python train...

**Comment by WeberJulian** [comment]
Also for glowTTS, in init_multispeaker for internal embeddings we have `self.embedded_speaker_dim = self.hidden_channels_enc`
For external embeddings, how do we deal with the fact that we have both d_vector_dim and hidden_channels_enc in the config? Should we remove d_vector_dim for glowTTS?

File: TTS/tts/datasets/dataset.py (hunk 1 above)
Comment (erogol) [reply TTS/tts/datasets/dataset.py:418]: I this the part that breaks d-vector training ?

Comment (WeberJulian) [reply TTS/tts/datasets/dataset.py:418]: yeah, `_sort_batch` already returns a sorted batch, when you pick the wav_file_name with the ids from the old order, it shuffles them compared to the rest of the items in the batch.

**Comment by erogol** [comment]
I think it is better to add a projection layer if d_vector_dim != hidden_channels_enc

**Comment by erogol** [comment]
> Should I add a `train_glow_tts_d-vector.py` recipe? If yes, does it means I need to resample and compute embedding in a subprocess in the recipe? Or have a main bash script that successively call resample -> compute emb -> python train...

it would be great to have that so we would have a better community feedback if something fails

**Comment by erogol** [comment]
do we need this or already fixed in  #1007

**Comment by WeberJulian** [comment]
Indeed, I applied this commit in the mutllilingual-vits branch, still interested in the recipe?

**Comment by erogol** [comment]
Then I close this and continue on the multilingual branch

**Pull request closed without merging**